### PR TITLE
added stub script that prints out latest commits

### LINKED ISSUE #76700: Migrate viable/strict promotion to GHA
Since our trunk CI is not the most reliable as of yet, we have the concept of a “viable/strict” branch where we enforce that every commit on the branch must pass all our CI checks. This branch is crucial for our build and release pipelines as we do not want to distribute broken binaries to our users. This branch is also useful for developers who want a guarantee that they will not inherit upstream failures while developing. 

The current way we update this branch is a janky internal cron job that is overly complicated, unreliable, and hard to debug. We want to migrate this functionality to an open source GitHub action so that:
1. The logic is clearer and easier to debug and thus more maintainable (better engineering!)
2. The code is reusable across all open source projects, such as our domain projects TorchVision and TorchAudio, etc. (scalability!)

The ask in this milestone is to create a GitHub action that would periodically promote the latest trunk commit where:
- All jobs are passing
- No jobs are pending
- At least 20 jobs have been run

Goal: 
A GitHub action that would periodically promote the latest green trunk commit to our viable/strict branch.

Resources:
- Understanding GHA: https://docs.github.com/en/actions/learn-github-actions/understanding-github-actions
- Pre-read our .github README: https://github.com/pytorch/pytorch/tree/master/.github/scripts 
- Internal code that the GHA will be replacing: https://www.internalfb.com/code/www/flib/intern/opensource/ci/OpensourceCIViability.php?lines=24 

Tasks:
- [x] Start a stub script that prints the latest commit SHAs on trunk from the last M minutes
- [x] Add a function that prints the check statuses run on a given commit SHA
- [x] Write stubbed out test cases for a function that would verify whether a commit SHA is promote-able
- [x] Write the function that would verify whether a commit SHA is promote-able
- [x] Write a GitHub Action that would run the completed script every 30 minutes and promote the most recent “green” commit SHA to viable/strict

### PR DESCRIPTION
**Pull request by swang392**
Relates to #76700

Prints out a list of commit SHAs from the last M minutes. Takes in a command line argument for the minutes, otherwise defaults to 60 mins.

**Test Plan**: compare output with the SHAs from the HUD and see if they are correct. Some SHAS printed with the script are not included on the HUD, but all of the commits on the HUD are outputted correctly.

### CODE REVIEW

File: .github/scripts/print_latest_commits.py [hunk 1]
```diff
@@ -0,0 +1,28 @@
+import sys
+import subprocess
+from datetime import datetime
+from datetime import timedelta
```
Comment (janeyx99) [inline .github/scripts/print_latest_commits.py:4]:
```suggestion
from datetime import datetime, timedelta
```

File: .github/scripts/print_latest_commits.py [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+            "--all",
+        ],
+        encoding="ascii",
+    ).splitlines()
+
+    for commit in commits:
+        print(commit)
+
+def main() -> None:
+    print_latest_commits() if len(sys.argv) == 1 else print_latest_commits(int(sys.argv[1]))
```
Comment (zengk95) [inline .github/scripts/print_latest_commits.py:24]: nit you can probably move the if else into the print_latest_commit function.

[the remark this answers is not in the document]
Comment (janeyx99) [reply .github/scripts/print_latest_commits.py:24]: Agreed that using the parse_args here would probably be clearest.

File: .github/scripts/print_latest_commits.py [hunk 3]
```diff
@@ -0,0 +1,34 @@
+from typing import Any
+from datetime import datetime, timedelta
+from gitutils import _check_output
+
+def parse_args() -> Any:
+    from argparse import ArgumentParser
+    parser = ArgumentParser("Print latest commits")
+    parser.add_argument("--minutes", type=int, help="duration in minutes of last commits")
```
Comment (seemethere) [inline .github/scripts/print_latest_commits.py:8]: Let's add a default here since I think this might be 1 or 0 by default?

**Review by seemethere — APPROVED** [review]
one more nit, but should be good to go

**Pull request closed without merging**